pkg/products/cloudresources/reconciler_test.go

```diff
@@ -240,5 +249,151 @@ func getBuildScheme() (*runtime.Scheme, error) {
 	if err != nil {
 		return nil, err
 	}
+	err = cloudcredentialv1.AddToScheme(scheme)
+	if err != nil {
+		return nil, err
+	}
+
 	return scheme, err
 }
+
+func TestReconciler_validateStsRoleArnPattern(t *testing.T) {
+
+	tests := []struct {
+		name          string
+		awsArnPattern string
+		want          bool
+	}{
+		{
+			name:          "ARN string Pattern is matching for AWS GovCloud (US) Regions",
+			awsArnPattern: "arn:aws-us-gov:iam::485026278258:role/12345",
+			want:          true,
+		},
+		{
+			name:          "ARN string Pattern is matching",
+			awsArnPattern: "arn:aws:iam::485026278258:role/12345",
+			want:          true,
+		},
+		{
+			name:          "ARN string Pattern is not matching #1",
+			awsArnPattern: "arn:aws:iam::485026278258:user/12345",
+			want:          false,
+		},
+		{
+			name:          "ARN string Pattern is not matching #2",
+			awsArnPattern: "12345",
+			want:          false,
+		},
+		{
+			name:          "ARN string Pattern is not matching #3",
+			awsArnPattern: "",
+			want:          false,
+		},
+	}
+	for _, tt := range tests {
+		got, err := validateStsRoleArnPattern(tt.awsArnPattern)
+		if err != nil {
+			t.Errorf("failed to validate STS role ARN parameter: %w", err)
+			return
+		}
+		if got != tt.want {
+			t.Errorf("validateStsRoleArnPattern() got = %v, want %v", got, tt.want)
+		}
+	}
+
+}
+
+func TestReconciler_passArnIntoSecretInCroNamespace(t *testing.T) {
+	scheme, err := getBuildScheme()
+	if err != nil {
+		t.Fatalf("Error obtaining scheme")
+	}
+	type args struct {
+		ctx    context.Context
+		client client.Client
+	}
+	tests := []struct {
+		name    string
+		ARN     string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "ARN Secret passed to Cro Namespace",
+			ARN:  "arn:aws:iam::485026278258:role/12345",
+			args: args{
+				ctx:    context.TODO(),
+				client: moqclient.NewSigsClientMoqWithScheme(scheme),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Method is not checking ARN pattern. Any ARN will be passed",
+			ARN:  "12345",
+			args: args{
+				ctx:    context.TODO(),
+				client: moqclient.NewSigsClientMoqWithScheme(scheme),
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		err := passArnIntoSecretInCroNamespace(tt.args.ctx, tt.args.client, operatorNamespace, tt.ARN)
+		if (err != nil) != tt.wantErr {
+			t.Errorf("passArnIntoSecretInCroNamespace() error = %v, wantErr %v", err, tt.wantErr)
+			return
+		}
```

Comment:
Here it would be good to check that secret is actually created.
You can do a .Get call with the mock client used in the test - `tt.args.client`
Example - https://github.com/integr8ly/integreatly-operator/blob/0bd1f9ea75a032a51eeb2d84ac070a0497aa3945/pkg/resources/pullSecret_test.go#L71-L76

Reply:
Done. Thank you for comment.